homework/index.js

```diff
@@ -15,32 +38,127 @@
     xhr.onerror = () => cb(new Error('Network request failed'));
     xhr.send();
   }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
+  // *********************************************************
+  function ShowContributions(urlText) {
+    const infoContainer = document.getElementById('info_container');
+    const infoRight = createAndAppend('div', infoContainer, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+    fetchJSON(urlText, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+          width: '50px',
+        });
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
     });
-    return elem;
   }
-
-  function main(url) {
+  // *************************************************
+  function ShowRepoInfo(arrayOfObjects, indexNumber) {
+    const wantedRepo = arrayOfObjects[indexNumber];
+    const infoContainer = document.getElementById('info_container');
+    infoContainer.innerHTML = '';
+    const infoLeft = createAndAppend('div', infoContainer, {
+      class: 'info-left',
+      text: 'Repository information:',
+    });
+    const nameAndLink = createAndAppend('p', infoLeft, {
+      class: 'p-info-left',
+      text: 'Name:',
+    });
+    createAndAppend('a', nameAndLink, {
+      class: 'repo-link',
+      text: wantedRepo.name,
+      href: wantedRepo.html_url,
+      target: '_blank',
+    });
+    const Description = createAndAppend('p', infoLeft, {
```

Comment:
Variable names should be camelCase: `description`.

Reply:
type mistake. I will fix it in the next  version.